docs/src/main/asciidoc/sql.adoc

```diff
@@ -90,6 +91,17 @@ You can select the type of connection pool (e.g., Tomcat, HikariCP, etc.) by htt
 Using the created `DataSource` in conjunction with Spring JDBC provides you with a fully configured and operational `JdbcTemplate` object that you can use to interact with your SQL database.
 You can connect to your database with as little as a database and instance names.
 
+==== Cloud SQL IAM database authentication
+
+Currently, Cloud SQL only supports https://cloud.google.com/sql/docs/postgres/authentication:[IAM database authentication for PostgreSQL].
+It allows you to connect to the database using an IAM account, rather than a predefined database username and password.
+You will need to do the following to enable it:
+
+. In your database instance settings, turn on the `cloudsql.iam_authentication` flag.
+. Add the IAM user or service account to the list of database users.
+. In the application settings, set `spring.cloud.gcp.sql.enableIamAuth` to `true`.
+Note that this will also set the `sslmode` to `disabled`, as it's required for IAM authentication to work.
```

Comment:
Is there Cloud docs we can point to? Hell of a thing to turn off SSL to make things more secure.

Reply:
> Does the feature exist on MySQL? If not, might want to clarify that we don't implement it due to it not being available.

The code change actually will also pass the property through for MySQL connections, but until MySQL gets IAM auth, it will do nothing. I believe it's coming to MySQL.

Reply:
I believe the connection is already encrypted, but just in case, I asked the question here: https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/pull/490#issuecomment-854152505